src/main/java/core/basesyntax/db/FruitShopStorage.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.OperationType;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Set;
+
+public class FruitShopStorage implements ShopStorage<Fruit> {
+    private static final String STORAGE_VALUE_ERROR = "Storage value cannot be negative";
+    private final Map<Fruit, Integer> fruitStorage = new EnumMap<>(Fruit.class);
```

Comment:
The `fruitStorage` is currently using `EnumMap`, which is not appropriate since `Fruit` is not an enum. Consider using a `HashMap` or another suitable `Map` implementation.